sentinel-extension/sentinel-datasource-consul/src/main/java/com/alibaba/csp/sentinel/datasource/consul/ConsulDataSource.java

```diff
@@ -0,0 +1,125 @@
+package com.alibaba.csp.sentinel.datasource.consul;
+
+import com.alibaba.csp.sentinel.concurrent.NamedThreadFactory;
+import com.alibaba.csp.sentinel.datasource.AbstractDataSource;
+import com.alibaba.csp.sentinel.datasource.Converter;
+import com.alibaba.csp.sentinel.log.RecordLog;
+import com.ecwid.consul.v1.ConsulClient;
+import com.ecwid.consul.v1.QueryParams;
+import com.ecwid.consul.v1.Response;
+import com.ecwid.consul.v1.kv.model.GetValue;
+
+import java.util.concurrent.*;
+
+import static java.util.concurrent.Executors.newSingleThreadExecutor;
+
+/**
+ * A read-only {@code DataSource} with Consul backend.
+ *
+ */
+public class ConsulDataSource<T> extends AbstractDataSource<String, T> {
+
+    private final ConsulClient client;
+
+    private final String address;
+
+    private String ruleKey;
+
+    private volatile long lastIndex;
+
+    private int watchTimeout;
+
+    private ConsulKVWatcher watcher = new ConsulKVWatcher();
+
+    private ExecutorService watcherService = newSingleThreadExecutor(
+            new NamedThreadFactory("sentinel-consul-ds-update", true));
+
+    public ConsulDataSource(Converter<String, T> parser, String host, int port, String ruleKey, int watchTimeout) {
```

Comment:
You have to provide the unit of the `watchTimeout` in the comment. It seems that the unit here should be `s` rather than commonly-used `ms`.

Reply:
And you may also provide a constructor with the default port (8500).